general/cuda.hpp

```diff
@@ -15,17 +15,15 @@
 #include "../config/config.hpp"
 #include "error.hpp"
 
-#ifdef MFEM_USE_CUDA
-#include <cuda_runtime.h>
-#include <cuda.h>
-#endif
-
 // CUDA block size used by MFEM.
 #define MFEM_CUDA_BLOCKS 256
 
 #ifdef MFEM_USE_CUDA
 #define MFEM_DEVICE __device__
+#define MFEM_LAMBDA __host__
```

Comment:
I would have found more consistent to call this `MFEM_HOST`, is there a reason to choose `MFEM_LAMBDA` over `MFEM_HOST`?

Reply:
`nvcc` only accepts `__host__` there, where `hipcc` requires `__host__ __device__` to work.
That's why I added this variant, to be able to pick the right one from the header.